File: ios/Classes/NativeViewFactory.swift
//
//  NativeViewFactory.swift
//  pubstar_io
//
//  Created by Mobile  on 13/6/25.
//

import Flutter
import UIKit

class NativeViewFactory: NSObject, FlutterPlatformViewFactory {
    func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> any FlutterPlatformView {
        let adView = NativeView(frame: frame, viewIdentifier: viewId)
        PubstarAdViewRegistry.shared.views[viewId] = adView.view()
        
        return adView
    }
}

File: android/src/main/kotlin/com/tqc/pubstar_io/NativeViewFactory.kt
package com.tqc.pubstar_io

import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class NativeViewFactory : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val adView = NativeView(context, viewId)
        PubstarAdViewRegistry.views[viewId] = adView.getAdContainer()

//        PubstarAdViewRegistry.views[viewId] = adView.view
        return adView
    }
}